src/Simplex/Messaging/Agent/Transmission.hs

```diff
@@ -218,7 +222,7 @@ newtype AckMode = AckMode Mode deriving (Eq, Show)
 data SMPQueueInfo = SMPQueueInfo SMPServer SMP.SenderId EncryptionKey
   deriving (Eq, Show)
 
-data ReplyMode = ReplyOff | ReplyOn | ReplyVia SMPServer deriving (Eq, Show)
+newtype ReplyMode = ReplyMode Mode deriving (Eq, Show)
```

Comment:
rename `Mode` to smth less generic, e.g. `OnOffMode`? At least this shows justification why both `ReplyMode` and `AckMode` are coupled to the same `Mode`